### PR TITLE
Remove dependency on graphdb-api in values

### PR DESCRIPTION
**Pull request by pontusmelke**
The `values` module was depending on `neo4j-graphdb-api` in order to be able to create wrapped node and relationship proxies. This dependency is now removed and the wrapping is now done in the top level cypher module instead of in the values module.

### CODE REVIEW

File: community/values/src/test/java/org/neo4j/values/storable/BufferValueWriter.java [hunk 1]
```diff
@@ -157,7 +156,7 @@ public void writeString( char value )
     @Override
     public void writeUTF8( byte[] bytes, int offset, int length ) throws RuntimeException
     {
-        buffer.add( UTF8.decode( bytes, offset, length ) );
+        buffer.add( new String( bytes, offset, length, StandardCharsets.UTF_8 ) );
```
Comment (fickludd) [inline community/values/src/test/java/org/neo4j/values/storable/BufferValueWriter.java:159]: Interesting... what is the reason?

**Review by fickludd — APPROVED** [review]
Looks good to me, this is a sensible restructuring to remove dependencies.

**Comment by fickludd** [comment]
[Should values depend on graphdb-api?](https://trello.com/c/77xJf6um/16-should-values-depend-on-graphdb-api)

**Comment by fickludd** [comment]
[Where should QueryResult live?](https://trello.com/c/vbOAF4GI/15-where-should-queryresult-live)

File: community/values/src/test/java/org/neo4j/values/storable/BufferValueWriter.java (hunk 1 above)
Comment (pontusmelke) [reply community/values/src/test/java/org/neo4j/values/storable/BufferValueWriter.java:159]: values doesn't doesn't have that utility as a dependency anymore. `UTF8.decode` just does that very thing.

**Pull request merged**